### PR TITLE
3-data

### PR DESCRIPTION
**Pull request by khurram-uworx**

### CODE REVIEW

File: Utils/Data/WorklogDictionary.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    Dictionary<string, string> lookup = new();
+
+    WorklogDictionary()
+    { }
+
+    public WorklogDictionary(DateTime month)
+    {
+        this.month = new DateTime(month.Year, month.Month, 1);
+        var nextMonth = this.month.AddMonths(1);
+        numberOfDays = nextMonth.Subtract(this.month).TotalDays;
```
Comment (khurram-juriba) [inline Utils/Data/WorklogDictionary.cs:21]: should be this.numberOfDays =

Comment (khurram-uworx) [inline Utils/Data/WorklogDictionary.cs:21]: should be this.numberOfDays =

File: Utils/Data/WorklogDictionary.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                    var seconds = this[k][i];
+                    var value = string.Empty;
+
+                    if (seconds >= 60 * 60)
+                        value = $"{seconds / 60.0 / 60.0:0.##}h";
+                    else if (seconds >= 60)
+                        value = $"{seconds / 60.0:0.##}m";
+                    else if (seconds > 0)
+                        value = $"{seconds}s";
+
```
Comment (khurram-uworx) [inline Utils/Data/WorklogDictionary.cs:63]: for zero; we want to show an empty cell

File: Utils/Helpers/JiraHelper.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    JiraHelper() { }
+
+    public JiraHelper(string uriString, string user, string password)
+    {
+        this.jiraClient = JiraClient.Create(new Uri(uriString));
+        this.jiraClient.SetBasicAuthentication(user, password);
+
+        using (var db = new JiraDbContext())
+        {
+            db.Database.EnsureCreated();
```
Comment (khurram-uworx) [inline Utils/Helpers/JiraHelper.cs:23]: this will created needed db structures

File: Utils/Messaging/MessageReceiver.cs [hunk 4]
```diff
@@ -61,6 +63,33 @@ public void ReceiveMessage(BasicDeliverEventArgs ea)
 
             activity?.SetTag("message", message);
 
+            var jiraUrl = Constants.JiraUrl;
+            if (!string.IsNullOrWhiteSpace(jiraUrl) && jiraUrl != "https://YOUR-COMPANY.atlassian.net")
+            {
+                try
+                {
+                    var jql = "Project = AppM AND worklogDate >= '2023-12-1' AND worklogDate < '2024-1-1'";
```
Comment (khurram-uworx) [inline Utils/Messaging/MessageReceiver.cs:71]: jql should come from environment variable as well

File: Utils/Data/Summarizer.cs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        var finish = start.AddMonths(1);
+        var dictionary = new WorklogDictionary(start);
+
+        using (var db = new JiraDbContext())
+        {
+            var emails = new[]
+            {
+                "abdul.hai@juriba.com", "abdulraffay.saeed@juriba.com", "arslan.ahmad@juriba.com", "hamza.mehmood@juriba.com",
+                "khurram.aziz@juriba.com", "mohammed.butt@juriba.com", "samia.saleem@juriba.com", "sana.fatehkhan@juriba.com", "tahreem.ahmad@juriba.com"
+            };
```
Comment (khurram-uworx) [inline Utils/Data/Summarizer.cs:23]: the list of our people can come from a txt file may be?

File: Utils/Data/Summarizer.cs [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+using System.Threading.Tasks;
+using Utils.Cache;
+
+namespace Utils.Data;
+
+internal class Summarizer
+{
+    public async Task<int> ProcessAsync()
+    {
+        var start = new DateTime(2023, 12, 1);
```
Comment (khurram-uworx) [inline Utils/Data/Summarizer.cs:13]: the month value can come from environment variable

**Pull request merged**